mono/utils/mono-hwcap-arm.c

```diff
@@ -150,6 +157,23 @@ mono_hwcap_arch_init (void)
 			}
 		}
 
+		/*
+		 * If we're executing as a 32-bit binary on a 64-bit system, we can
+		 * safely assume the presence of all 32-bit features.
+		 */
+		if (arm64) {
+			mono_hwcap_arm_is_v5 = TRUE;
+			mono_hwcap_arm_is_v6 = TRUE;
+			mono_hwcap_arm_is_v7 = TRUE;
+
+			mono_hwcap_arm_has_vfp = TRUE;
+			mono_hwcap_arm_has_vfp3 = TRUE;
+			mono_hwcap_arm_has_vfp3_d16 = TRUE;
+
+			mono_hwcap_arm_has_thumb = TRUE;
+			mono_hwcap_arm_has_thumb2 = TRUE;
+		}
```

Comment:
Judging by the instructions listed in the AArch32 section of the ARMv8 manual, I'd say assuming ARMv7 is safe. Judging by E1.2.1 and E1.3.2, I'd say assuming full VFPv3 support is also safe.

Thumb I'm not actually sure about. Any idea?